src/NodaTime.Test/ConsistencyTest.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright 2020 The Noda Time Authors. All rights reserved.
+// Use of this source code is governed by the Apache License 2.0,
+// as found in the LICENSE.txt file.
+
+using NodaTime.TimeZones.Cldr;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace NodaTime.Test
+{
+    /// <summary>
+    /// Tests using reflection to check that types are consistent.
+    /// </summary>
+    public class ConsistencyTest
+    {
+        private static IEnumerable<Type> AllPublicTypes = typeof(LocalDate).Assembly
+            .GetTypes()
+            .Where(t => t.IsPublic)
+            .ToList();
+
+        [Test]
+        public void EqualityOperatorForEquatable()
+        {
+            var expectedExceptions = new[] { typeof(MapZone) };
+
+            var failures = AllPublicTypes
+                .Where(t => typeof(IEquatable<>).MakeGenericType(t).IsAssignableFrom(t))
+                .Where(t => t.GetMethod("op_Equality") is null)
+                .Except(expectedExceptions)
+                .ToList();
+            Assert.IsEmpty(failures);
+        }
+    }
```

Comment:
At the risk of being somewhat pedantic, how about an additional test for `IComparable` types having all of {`<`, `<=`, `>=`, `>`} too?

Reply:
Hmm. I think we're likely to need a lot more exceptions for this, as it's quite reasonable to make things comparable without overloading those operators (e.g. string) but I'll try adding the test anyway - if we only need a few exceptions, it wouldn't hurt to have a reminder.